tests/osaka/eip7692_eof_v1/eip5450_stack/test_code_validation.py

```diff
@@ -561,9 +561,63 @@ def test_all_opcodes_stack_underflow(
                 ),
             ],
         ),
+        Container(
+            name="underflow_variable_stack_3",
+            sections=[
+                Section.Code(
+                    Op.RJUMPI[8](0)
+                    + Op.PUSH0 * 2
+                    + Op.RJUMPI[1](0)
+                    + Op.POP * 2
+                    + Op.PUSH0 * 2
+                    + Op.REVERT,
+                    max_stack_height=4,
+                ),
+            ],
+        ),
     ],
     ids=lambda x: x.name,
 )
 def test_stack_underflow_examples(eof_test, container):
     """Test EOF validation failing due to stack underflow at basic instructions."""
     eof_test(container=container, expect_exception=EOFException.STACK_UNDERFLOW)
+
+
+@pytest.mark.parametrize(
+    "container",
+    [
+        Container(
+            name="non_constant_stack_height_0",
+            sections=[
+                Section.Code(
+                    code=Op.RJUMPI[8](0)
+                    + Op.PUSH0 * 2
+                    + Op.RJUMPI[1](0)
+                    + Op.POP
+                    + Op.PUSH0 * 2
+                    + Op.REVERT,
+                    max_stack_height=4,
+                ),
+            ],
+        ),
+        Container(
+            name="non_constant_stack_height_1",
+            sections=[
+                Section.Code(
+                    code=Op.PUSH0
+                    + Op.RJUMPI[8](0)
+                    + Op.PUSH0 * 2
+                    + Op.RJUMPI[1](0)
+                    + Op.POP
+                    + Op.PUSH0 * 2
+                    + Op.REVERT,
+                    max_stack_height=5,
+                ),
+            ],
+        ),
+    ],
+    ids=lambda x: x.name,
+)
+def test_valid_non_constant_stack_examples(eof_test, container):
```

Comment:
Tests with valid EOF work better as `eof_state_test()` so maybe at least add a TODO to convert them later.

Reply:
I converted the test to EOFStateTestFiller